### ISSUE TITLE
Metallb v0.11.0 + RHOS 4.8.22

### ISSUE THREAD

**Issue by dblviking** [metallb/metallb#1213]
I'm having issue with connectivity with my ipv6 metallb addresses. The services are correctly pulling an ip from my declared layer2 ipv6 pool. Everything seems fine but I can't connect to them.

My metallb controller correctly setup an ipv4 and ipv6 client address but the speakers only show an ipv4 worker node ip, not a client address. I'm curious if this is the issue? I'm not sure how the speaker works in a dualstack setup.

**Comment by fedepaol** [comment]
Hi @dblviking , is this L2 or bgp mode?
Please note that ping won't work in either way, because no interface on the node is holding the vip.

**Comment by dblviking** [comment]
It’s L2. Yes you’re right about ping. I started using nc to check for the open ports. I get “no route to hosts”

**Comment by fedepaol** [comment]
Is your service on the same subnet of the client? In L2 mode the client and the server are supposed to be on the same subnet, so routing should work out of the box.

**Comment by dblviking** [comment]
Yes I have a nmstate vlan on the worker interfaces and they can all ping6 each other and the gateway.  They’re all on the same subnet. 
My ipv4 install worked fine but this is puzzling. 
is there any internal metallb ingress routing I can check?

**Comment by dblviking** [comment]
It's of note that the metallb-controller has dual stack ipv4 and ipv6 address but the speaker pods are hard coded with worker node internal ipv4 addresses only (these ones: oc get nodes -o wide)

**Comment by dblviking** [comment]
oc describe pod controller-8fb84f97d-wvtzf -n metallb-system | grep IP
IP:           10.9.6.67
IPs:
  IP:           10.9.6.67
  IP:           fd01:0:0:6::1468

oc describe pod speaker-6blxw -n metallb-system | grep IP
IP:           <ipv4 worker0 node ip>   <--- shouldn't that be ipv6 or both?
IPs:
  IP:           <ipv4 worker0 node ip><--- shouldn't that be ipv6 or both?
      METALLB_HOST:            (v1:status.hostIP)
      METALLB_ML_BIND_ADDR:    (v1:status.podIP)

**Comment by dblviking** [comment]
How do I force the speaker pods to use the ipv6 of the node, not the ipv4?

oc describe pod -n metallb-system speaker-6blxw
Name:         speaker-6blxw
Namespace:    metallb-system
Priority:     0
Node:         worker0/10.10.10.73
Start Time:   Tue, 08 Feb 2022 13:50:55 -0500
Labels:       app=metallb
              component=speaker
              controller-revision-hash=59cbb56d76
              pod-template-generation=1
Annotations:  openshift.io/scc: privileged
              prometheus.io/port: 7472
              prometheus.io/scrape: true
Status:       Running
IP:           10.10.10.73
IPs:
  IP:           10.10.10.73
Controlled By:  DaemonSet/speaker

**Comment by fedepaol** [comment]
So, metallb does not do any routing. In l2 the only thing it does is to reply to your arp requests. All the rest is done by the cluster.
If you see arp requests coming to the node, metallb's job is done.

What version of ocp are you using? And also, is this the community metallb or the one shipped with openshift? If the latter, I think you can file a proper bug / incident request.

**Comment by fedepaol** [comment]
PS: The speaker pods are host-networked pods, so they take the same ip of the node. Not sure why it's rendering only the v4 one, but if your nodes are dual stack, so is the speaker pod.

**Comment by fedepaol** [comment]
Closing this as there was no feedback on the questions, feel free to reopen if you want to provide the missing information.
